source/statements.tex

```diff
@@ -315,6 +317,57 @@
 except that the \grammarterm{init-statement} is
 in the same scope as the \grammarterm{condition}.
 
+\pnum
+An \keyword{if} statement of the form \tcode{\keyword{if} \keyword{consteval}}
+is called a \defnadj{consteval if}{statement}.
+The \grammarterm{statement}, if any, in a consteval if statement
+shall be a \grammarterm{compound-statement}.
+\begin{example}
+\begin{codeblock}
+constexpr void f(bool b) {
+  if (true)
+    if consteval { }
+    else ;              // error: not a \grammarterm{compound-statement}; \keyword{else} not associated with outer \keyword{if}
+}
+\end{codeblock}
+\end{example}
+
+\pnum
+If a consteval if statement is evaluated in a context
+that is manifestly constant-evaluated\iref{expr.const},
+the first substatement is executed.
+\begin{note}
+The first substatement is an immediate function context.
+\end{note}
+Otherwise, if the \keyword{else} part of the selection statement is present,
+then the second substatement is executed.
+A \keyword{case} or \keyword{default} label
+appearing within such an \keyword{if} statement
+shall be associated with a \keyword{switch} statement
+within the same \keyword{if} statement.
+A label declared in a substatement of a consteval if statement
+shall only be referred to by a statement in the same substatement.
+
+\pnum
+An \keyword{if} statement of the form
+\begin{ncsimplebnf}
+\keyword{if} \terminal{!} \keyword{consteval} compound-statement
+\end{ncsimplebnf}
+is not itself a consteval if statement,
+but is equivalent to the consteval if statement
+\begin{ncsimplebnf}
+\keyword{if} \keyword{consteval} \terminal{\{} \terminal{\}} \keyword{else} compound-statement
+\end{ncsimplebnf}
+An \keyword{if} statement of the form
```

Comment:
I'm not loving the fact that this sentence doesn't end properly, but that's what's in the paper...